File: backend/app/schemas/payment.py
"""Pydantic schemas for payment and subscription."""

from datetime import datetime
from enum import Enum

from pydantic import BaseModel


class PayChannel(str, Enum):
    """Supported payment channels."""

    ALIPAY = "alipay"
    WECHAT = "wechat"


class CreateOrderRequest(BaseModel):
    """Request body for creating a payment order."""

    pay_channel: PayChannel
    duration_months: int = 1


class CreateOrderResponse(BaseModel):
    """Response after creating a payment order."""

    order_id: str
    out_trade_no: str
    qr_code_url: str
    amount: int
    pay_channel: str


class OrderStatusResponse(BaseModel):
    """Response for order status query."""

    order_id: str
    status: str
    pay_channel: str
    amount: int
    paid_at: datetime | None = None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/api/payment.ts
import client from './client'

export interface CreateOrderParams {
  pay_channel: 'alipay' | 'wechat'
  duration_months: number
}

export interface CreateOrderResult {
  order_id: string
  out_trade_no: string
  qr_code_url: string
  amount: number
  pay_channel: string
}

export interface OrderStatus {
  order_id: string
  status: string
  pay_channel: string
  amount: number
  paid_at: string | null
  created_at: string
}

export const paymentApi = {
  createOrder: async (params: CreateOrderParams): Promise<CreateOrderResult> => {
    const { data } = await client.post('/payment/create', params)
    return data
  },

  getOrderStatus: async (orderId: string): Promise<OrderStatus> => {
    const { data } = await client.get(`/payment/status/${orderId}`)
    return data
  },
}
